src/dg/strong_dg.cpp

```diff
@@ -1740,25 +2093,11 @@ void DGStrong<dim,nstate,real,MeshType>::assemble_face_term_strong(
     // the outward reference normal dircetion.
     const dealii::Tensor<1,dim,double> unit_ref_normal_int = dealii::GeometryInfo<dim>::unit_normal_vector[iface];
     // Extract the reference direction that is outward facing on the facet.
-    int dim_not_zero_probe;
-    if constexpr(dim==1){
-      dim_not_zero_probe = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 : 1000;
-    }else if constexpr(dim==2){
-      dim_not_zero_probe = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 
-                     : (abs(unit_ref_normal_int[1]) >= 1e-11 ? 1 : 1000);
-    }else if constexpr(dim==3){
-      dim_not_zero_probe = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 
-                         : (abs(unit_ref_normal_int[1]) >= 1e-11 ? 1 
-                         : (abs(unit_ref_normal_int[2]) >= 1e-11 ? 2 : 1000));
-    }else{
-      pcout<<"Error with normals. Assume dim <=3. Aborting..."<<std::endl;
-      std::abort();
-    }
-    const int dim_not_zero = dim_not_zero_probe;
-    if(dim_not_zero == 1000){
-        pcout<<"Error with normals. Normal direction is not defined. Aborting..."<<std::endl;
-        std::abort();
-    }
+    const int dim_not_zero = iface / 2;//reference direction of face integer division
+//    if(dim_not_zero == 1000){
+//        pcout<<"Error with normals. Normal direction is not defined. Aborting..."<<std::endl;
+//        std::abort();
+//    }
```

Comment:
do we still need the check?

Reply:
No because I found a simpler way to extract the dim_not_zero directly.